src/aleph_client/asynchronous.py

```diff
@@ -562,6 +562,36 @@ async def get_posts(
         return await resp.json()
 
 
+async def get_store_file(
+        file_hash: str,
+        storage_engine: Optional[StorageEnum] = None,
+        session: Optional[ClientSession] = None,
+        api_server: Optional[str] = None
+) -> bytes:
```

Comment:
Beware of large files, this function might end up using a lot of memory.

Reply:
Recommend some type of generator then? Added a warning for starters ^^'

Reply:
A warning is a good start. A bytes generator might be a good idea, as long as what the user is doing with it does not take too long. And then adding higher-level functions on top of the generator, like `def download_file(file_hash: str, output_file: pathlib.Path)` etc.

Reply:
Judging from https://github.com/aleph-im/pyaleph/blob/190ebef84b5823c4ea5616f5152b4f6fa21b17b9/src/aleph/web/controllers/storage.py#L79
there is currently no way to request a StreamingResponse, so we inadvertently will have the whole file in memory until we store/dispose of it in a file, correct?

Reply:
Indeed, but that is on the server side with lots of RAM.

Reply:
Returning a file descriptor may be ideal. 

Headers contain the size of the content, which could be stored in a temporary file when larger than a certain threshold.

Reply:
See https://docs.python.org/3/library/io.html#io.BufferedReader

Reply:
Let's create an issue for this then :+1: